providers/common/digests/sha3.c

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/sha.h>
+#include <openssl/crypto.h>
+#include <openssl/core_numbers.h>
+#include <openssl/evp.h>
+
+static int shake_ctrl(KECCAK1600_CTX *ctx, int cmd, int p1, void *p2)
```

Comment:
I thoroughly dislike that we reproduce the old ctrl() signatures like this, and would rather see a construction with stdargs.

However, this could also be handled with a `set_params` function that takes an `OSSL_PARAM` array...

Reply:
Ctrl functions like the dodgy one in SHA1 are running code (as opposed to SHAKE which just sets a  parameter). Shouldnt be too hard to change it to set_params if that is what people think it should be?

Reply:
I see no problem whatsoever in using OSSL_PARAMs for triggering purposes.
(regarding the dodgy SHA1 control code, I'm of the mind that stuff should be re-implemented in a cleaner way, and perhaps no necessarily in a provider)